docs/semantics-of-k.tex

```diff
@@ -2287,6 +2294,95 @@ \subsection{Rewriting and Reachability}
 prove all the proof rules of reachability logic as lemmas/theorems using the
 more basic proof system of matching logic.
 
+\subsection{An Example}
+
+Consider the following example.
+Suppose we want to define a simple transition system
+whose state/configuration set is $\nats \cup \{ \rand \}$,
+where $\rand$ is the program construct representing a random number generator.
+The intended semantics of $\rand$ is that it can go to any number $n \in \nats$
+in one step.
+How can we write an axiom that captures the intended semantics?
+
+We here give three candidate axioms.
+\begin{align*}
+& \prule{Axiom A} &&
+\forall x . (\rand \imp (x > 0 \imp \snext(x)))
+&& \text{\doubleslash correct} \\
+& \prule{Axiom B} &&
+\rand \imp \exists x . (x > 0 \wedge (\snext x))
+&& \text{\doubleslash incorrect} \\
+& \prule{Axiom C} &&
+\forall x . (\rand \imp (x > 0 \wedge \snext(x)))
+&& \text{\doubleslash totally wrong}
+\end{align*}
+
+We first explain why \prule{Axiom B} and \prule{Axiom C} are wrong.
+First of all, \prule{Axiom C} is \emph{inconsistent}, meaning that
+we can prove everything, even ``bottom $\bot$'', from \prule{Axiom C}.
+To see why, we simply instantiate the ``$\forall x$'' with $x = 0$,
+and then we get $\forall x . \rand \imp (0 > 0 \wedge \snext(x))$.
+Since $0 > 0$ is false, meaning it equals $\bot$,
+we get
+$\rand \imp \bot$.
+
+\prule{Axiom B} is also wrong, but it is \emph{consistent}.
+What it says is that $\rand$ must be the predecessor of \emph{some positive 
+number}.
+Therefore, the transition system where $\rand$ can \emph{only} go to,
+say, $42$, satisfies \prule{Axiom B}.
+This means that \prule{Axiom B} is too weak to capture the intended semantics 
+of $\rand$.
+
+\prule{Axiom A} is the correct axiom. It says that \emph{for all positive 
+numbers 
+$n$},
+$\rand$ is a predecessor of $n$.
+This captures precisely the intended semantics of $\rand$.
+Note that \prule{Axiom A} is a \emph{positive axiom}, meaning that
+it merely says what the transition system \emph{must do},
+but not constrains what it \emph{cannot do}.
+Positive axioms will \emph{not} introduce inconsistency, as 
+the total transition system (i.e., the translation relation is the total 
+relation and all states can go to all other states, including themselves)
+is always a candidate model.
+
+To conclude this example, we show that if we want to capture \emph{in further}
+that $\rand$ can \emph{only} go to positive numbers and nowhere else,
+we need the following axiom, which is \emph{not} a positive axiom anymore:
+$$
+\prule{Axiom D} \qquad \rand \imp \wnext ( \exists x . (x > 0) \wedge x )
+$$
+where $\wnext \varphi \equiv \neg \snext \neg \varphi$ is the dual
+of ``one-path next'' $\snext$, called ``all-path next''. 
+The axiom says that \emph{for all next states} of $\rand$,
+it must be some positive number.
+\prule{Axiom A} and \prule{Axiom D} captures the intended semantics
```

Comment:
Did you mean to include "Axiom A" here? The reminder of the sentence seems to imply a single axiom (e.g. "captures" instead of "capture")

Reply:
That's a typo. Thanks for pointing out. It should be "capture".